plugins/inputs/logfile/tail/tail_test.go

```diff
@@ -71,22 +76,46 @@ func TestNotTailedCompeletlyLogging(t *testing.T) {
 	verifyTailerExited(t, tail)
 }
 
-func TestDroppedLinesWhenStopAtEOFLogging(t *testing.T) {
-	tmpfile, tail, tlog := setup(t)
+func TestStopAtEOF(t *testing.T) {
+	tmpfile, tail, _ := setup(t)
 	defer tearDown(tmpfile)
 
 	readThreelines(t, tail)
 
-	// Ask the tailer to StopAtEOF
-	tail.StopAtEOF()
+	// Since StopAtEOF() will block until the EOF is reached, run it in a goroutine.
+	done := make(chan bool)
+	go func() {
+		tail.StopAtEOF()
+		close(done)
+	}()
+
+	// Verify the goroutine is blocked indefinitely.
+	select {
+	case <-done:
+		t.Fatalf("StopAtEOF() completed unexpectedly")
+	case <-time.After(time.Second * 1):
```

Comment:
Waiting 1 second is arbitrary.
It could be increased, but I am trying not to increase unit test execution time _too much_.

Reply:
Just to confirm, would the channel ever get closed if we waited long enough? 🤔  I'm fine with this, just wanted to see.

Reply:
Channel should not close until `StopAtEOF()` returns, and that should not happen until all lines are read from the file.
`setup()` writes 10 lines, and this test case has only read 3 so far.